packages/web-components/fast-foundation/src/number-field/number-field.ts

```diff
@@ -249,12 +249,20 @@ export class FASTNumberField extends FormAssociatedNumberField {
      * @internal
      */
     private getValidValue(value: string): string {
-        let validValue: number | string = parseFloat(parseFloat(value).toPrecision(12));
-        if (isNaN(validValue)) {
+        let validValue: string = value;
+        if (isNaN(parseFloat(value))) {
             validValue = "";
         } else {
-            validValue = Math.min(validValue, this.max ?? validValue);
-            validValue = Math.max(validValue, this.min ?? validValue).toString();
+            const numericValue: number = parseFloat(parseFloat(value).toPrecision(12));
+            const minValue: number = this.min !== undefined ? this.min : numericValue;
+            const maxValue: number = this.max !== undefined ? this.max : numericValue;
+
+            // Check if the value falls within the valid range
+            if (numericValue < minValue) {
+                validValue = minValue.toString();
+            } else if (numericValue > maxValue) {
+                validValue = maxValue.toString();
+            }
```

Comment:
Your code also accepts numbers like this: '5-', which are not supposed to be valid.

You are missing an `else` in the final conditional:

```suggestion
            if (numericValue < minValue) {
                validValue = minValue.toString();
            } else if (numericValue > maxValue) {
                validValue = maxValue.toString();
            } else {
                validValue = numericValue.toString();
            }
```